hail/src/main/scala/is/hail/io/fs/GoogleStorageFS.scala

```diff
@@ -123,67 +125,60 @@ class GoogleStorageFS(
 
   def urlAddPathComponent(url: URL, component: String): URL = url.addPathComponent(component)
 
-  def getConfiguration(): Option[RequesterPaysConfiguration] =
-    requesterPaysConfiguration
+  def getConfiguration(): Option[RequesterPaysConfig] =
+    requesterPaysConfig
 
   def setConfiguration(config: Any): Unit =
-    requesterPaysConfiguration = config.asInstanceOf[Option[RequesterPaysConfiguration]]
+    requesterPaysConfig = config.asInstanceOf[Option[RequesterPaysConfig]]
 
   private[this] def requesterPaysOptions[T](bucket: String, makeUserProjectOption: String => T)
-    : Seq[T] = {
-    requesterPaysConfiguration match {
-      case None =>
-        Seq()
-      case Some(RequesterPaysConfiguration(project, None)) =>
+    : Seq[T] =
+    requesterPaysConfig match {
+      case Some(RequesterPaysConfig(project, buckets)) if buckets.fold(true)(_.contains(bucket)) =>
         Seq(makeUserProjectOption(project))
-      case Some(RequesterPaysConfiguration(project, Some(buckets))) =>
-        if (buckets.contains(bucket)) {
-          Seq(makeUserProjectOption(project))
-        } else {
-          Seq()
-        }
+      case _ =>
+        Seq()
     }
-  }
 
-  private[this] def retryIfRequesterPays[T, U](
-    exc: Throwable,
-    makeRequest: Seq[U] => T,
-    makeUserProjectOption: String => U,
-    bucket: String,
-  ): T =
-    if (isRequesterPaysException(exc)) {
-      makeRequest(requesterPaysOptions(bucket, makeUserProjectOption))
-    } else {
-      throw exc
-    }
+  object RequesterPaysFailure {
+    def unapply(t: Throwable): Option[Throwable] =
```

Comment:
I get that this is idomatic, but it feels too magical to me right now.

Reply:
Too magical? Would you like me to revert?

Reply:
not at all